### ISSUE TITLE
startReached + endless scrolling up? 

### ISSUE THREAD

**Issue by uffou** [petyosi/react-virtuoso#107]
Are you planning to add endless scrolling up? Something similar to endReached could be startReached. I am going to try your library for a chat I am building and I like it way better than react-virtualized so far. 

(Страхотна работа!)

**Comment by petyosi** [comment]
Thank you! 

Warning: scrolling upwards does not work well with variable size items. Displacements happen due to items not matching their expected size, and pushing down or pulling up previously visible ones. I have tried a few ways to fix it, but it needs different rendering, which also has some possible shortcomings. 

Up to your question, you can use `rangeChanged` and check when the startIndex gets close to zero in order to load the previous items.

**Comment by uffou** [comment]
I ended up using custom ScrollContainer where I hook up to the onScroll callback and then I use
clientHeight, scrollHeight, and scrollTop to load more items on top or bottom.

```
const goingUp = scrollTop < this.scrollTop
const goingDown = !goingUp
const nearStart = scrollTop < (clientHeight / 2)
const nearEnd = scrollTop > scrollHeight - (1.5 * clientHeight)

if (goingUp && nearStart) {
	console.log('Load more top')
} else if (goingDown && nearEnd) {
	console.log('Load more bottom')
}
```
I will try `rangeChanged` if I have any problems with the current approach.

Yes, there are some glitches scrolling up. If there was a boolean like to enable calculating those heights and then prepending those items I would use that without hesitation.

**Comment by chrisdostert** [comment]
having this built in would be super handy

**Comment by petyosi** [comment]
startReached is available in `v0.20.0`
